src/mlpack/methods/ann/layer/repeat_impl.hpp

```diff
@@ -152,24 +150,24 @@ void RepeatType<MatType>::ComputeOutputDimensions()
     }
     outSize *= this->outputDimensions[i];
   }
-  outIdxs.resize(idxs.n_elem);
-  for (size_t i=0; i<idxs.n_elem; i++)
+  outIdxs = idxs.as_col();
+
+  // Now, we are going to pre-compute the contribution of each output
+  // element to the input elements.  This will be used in the backward
+  // pass with a simple matrix multiplication.
+  backIdxs.set_size(inputSize, sizeMult);
+  UintCol counts(inputSize, arma::fill::zeros);
+  for (size_t i = 0; i < outIdxs.n_elem; i++)
   {
-    outIdxs[i] = idxs.at(i);
+    auto r = outIdxs.at(i);
```

Comment:
If this is always `arma::uword`, I'd just use that here.

Reply:
:+1: